rke/templates/metrics_v0.5.0.go

```diff
@@ -213,6 +213,7 @@ spec:
           readOnlyRootFilesystem: true
           runAsNonRoot: true
           runAsUser: 1000
+          allowPrivilegeEscalation: false
```

Comment:
I'm not sure we're OK changing existing templates and not introducing a new one.
@kinarashah @Oats87 , is this OK here?

Reply:
We cannot do this, we need to introduce a new version of the template

Reply:
@Oats87 , thank you.

@HarrisonWAffel , could you please do this? It's also a good timing as we've just merged KDM changes for August k8s patches so can just update newly added versions. We may still want to introduce new `rancherX-(Y+1)` versions for existing k8s versions where we're not adding new patches as part of August patches release.
This depends on which k8s versions we need this fixed for.

Unless it's not worth it for such a small change (I mean introducing a whole new template for a boolean field). I've heard from @kinarashah that "technically it'll be fine but I want to go with the pattern we've set and add a new template".

LMK if we want to discuss it further.